server/py/services/api/initial_data.py

```diff
@@ -132,22 +134,26 @@ def _initialize_db_from_scratch(
 
 
 def _migrate_existing_data(
+    engine: sqlalchemy.engine.Engine,
     perform_migrations_if_needed: bool = False,
 ):
-    # create mysql util, and if mlrun is configured to use mysql, wait for it to be live and set its db modes
-    mysql_util = framework.utils.db.mysql.MySQLUtil(
-        mlrun.utils.logger
-    )  # TODO make this DB agnostic
     alembic_util = _create_alembic_util()
-    if mysql_util.get_mysql_dsn_data():
-        mysql_util.wait_for_db_liveness()
-        mysql_util.set_modes(mlrun.mlconf.httpdb.db.mysql.modes)
+    if engine.dialect in (
+        mlrun.common.db.dialects.Dialects.MYSQL,
+        mlrun.common.db.dialects.Dialects.POSTGRESQL,
+    ):
+        db_util = DBUtil()
+        db_util.wait_for_db_liveness()
+        if engine.name == mlrun.common.db.dialects.Dialects.MYSQL:
+            db_util.set_configurations(
+                config_items=mlrun.mlconf.httpdb.db.mysql_engine.modes,
```

Comment:
Did you mean this? Because `mysql_engine` does not exist in the mlrun config
```suggestion
                config_items=mlrun.mlconf.httpdb.db.mysql.modes,
```